### PR TITLE
(cloudwatch) annotation support

### PR DESCRIPTION
**Pull request by mtanda**
CloudWatch store alarm(alert) history in their storage.
We can get the history by DescribeAlarmHistory.
http://docs.aws.amazon.com/AmazonCloudWatch/latest/APIReference/API_DescribeAlarmHistory.html

So, I suggest to use the history for annotation data source.
cloudwatch_annotation

### CODE REVIEW

**Comment by torkelo** [comment]
is there anyway we can reuse the query editor or parts of it? seems very similar

**Comment by mtanda** [comment]
@torkelo Reuse part of query.editor.html and query_ctrl.js?
I'm not sure, but I think it is possible.
To reuse the query.editor.html, I need to change current UI, and it might be little ugly.

May be, I need to combine these part of HTML.
https://github.com/grafana/grafana/blob/4dcd2ceb0124f2667ef2bcb053a320398d6c24d3/public/app/plugins/datasource/cloudwatch/partials/query.editor.html#L35-L54
https://github.com/grafana/grafana/blob/4dcd2ceb0124f2667ef2bcb053a320398d6c24d3/public/app/plugins/datasource/cloudwatch/partials/query.editor.html#L60-L67
https://github.com/grafana/grafana/blob/4dcd2ceb0124f2667ef2bcb053a320398d6c24d3/public/app/plugins/datasource/cloudwatch/partials/query.editor.html#L81-L87

On the other hand, the annotation editor will support suggestion.

**Comment by mtanda** [comment]
If you permit to change the UI, I'll try to reuse some of code.

**Comment by torkelo** [comment]
maybe reuse the whole query editor component, might require some big refeactorings.

**Comment by mtanda** [comment]
hmm, I may understand...
Let me think about it how I can do it.

**Comment by mtanda** [comment]
rebase and import this change 3de4707#diff-ecfe215a9820d2930d1157cca99c0acfR63 .

**Comment by mtanda** [comment]
rebase and import this change https://github.com/grafana/grafana/pull/3162

**Comment by mtanda** [comment]
@torkelo Finally, I reuse most of the query editor. Now, ready to review.
> is there anyway we can reuse the query editor or parts of it? seems very similar

**Comment by bergquist** [comment]
@mtanda Would you please solve the conflicts?

**Comment by mtanda** [comment]
@bergquist Sure! I resolve conflict :-)

**Comment by bergquist** [comment]
Great! I will test and confirm this asap.

File: public/app/plugins/datasource/cloudwatch/partials/query.editor.html [hunk 1]
```diff
@@ -33,61 +33,16 @@
...
-		<li>
-			<input type="text" class="input-mini tight-form-input" ng-model="target.period" spellcheck='false' placeholder="auto" ng-model-onblur ng-change="refreshMetricData()" />
-		</li>
-
 	</ul>
 
 	<div class="clearfix"></div>
 </div>
 
+<cloudwatch-query-parameter target="target" datasource-name="{{datasource.name}}" on-change="refreshMetricData()"></cloudwatch-query-parameter>
```
Comment (torkelo) [inline public/app/plugins/datasource/cloudwatch/partials/query.editor.html]: does this work with the `-- Mixed --` data source In that mode the data source is specified on each target. Maybe it does, test it :)

Comment (bergquist) [inline public/app/plugins/datasource/cloudwatch/partials/query.editor.html]: It works.

Comment (torkelo) [inline public/app/plugins/datasource/cloudwatch/partials/query.editor.html]: ok great. Then I do not think I have any problem merging this.

**Comment by mtanda** [comment]
Wow, thanks for merging!

**Pull request merged**